pkg/stenciltest/stenciltest.go

```diff
@@ -112,17 +113,55 @@ func (t *Template) ErrorContains(msg string) {
 	t.errStr = msg
 }
 
+// getTemplateRepositoryNames retrieves the naems from the passed in TemplateRepositories
+func getTemplateRepositoryNames(trs []*configuration.TemplateRepository) []string {
+	deps := make([]string, len(trs))
+	for i, tr := range trs {
+		deps[i] = tr.Name
+	}
+	return deps
+}
+
+// modulesFromTemplateRepository creates new modules from the passed in TemplateRepositories
+func modulesFromTemplateRepository(trs []*configuration.TemplateRepository) ([]*modules.Module, error) {
+	var mods []*modules.Module
+	for _, tr := range trs {
+		// Use the present version or main
+		if tr.Version == "" {
+			tr.Version = "main"
```

Comment:
I think if `tr.Version` is left undefined it'll determine the latest version to use... I don't like assuming the default branch here

Reply:
It doesn't automatically determine it. Let me see if I can refactor some of the version resolution code so it can be used here.